lib/kitchen/loader/yaml.rb

```diff
@@ -207,11 +211,19 @@ def dot_kitchen_yml
 
       # Determines the default absolute path to the Kitchen local YAML file,
       # based on the base Kitchen config YAML file.
-      #
+
       # @return [String] an absolute path to a Kitchen local YAML file
       # @api private
       def default_local_config_file
-        config_file.sub(/(#{File.extname(config_file)})$/, '.local\1')
+        default_local_config = config_file.sub(/(#{File.extname(config_file)})$/, '.local\1')
+
+        unhidden_config = default_local_config.sub(/([\/])\.([^\/]*)$/, '\1\2')
+        hidden_config = default_local_config.sub(/([\/])([^\.][^\/]*)$/, '\1.\2')
```

Comment:
I'm not proud of this.

Reply:
How about this?

```ruby
localize = lambda path: path.sub(/(#{File.extname(path)})$/, '.local\1')
if config_file == default_config_file && File.exist?(localize.call(kitchen_yml) && File.exist?(localize.call(dot_kitchen_yml)
```

or something like that?

Reply:
I like it. I had a phase of this change that had a whole method to do the `.local`izing, but an anonymous lambda in here is more better.

Reply:
Ah. Issue: by calling `kitchen_yml` and `dot_kitchen_yml` here, it only tests the existence of `kitchen.local.yml` and `.kitchen.local.yml`, but does not catch the case when `KITCHEN_YAML` is set to something and the `.local` version is computed based on that.

For example:

```
> export KITCHEN_YAML=asparagus.yml
> ls *.yml
asparagus.yml
asparagus.local.yml
.asparagus.local.yml
```

Ought to error just as when `kitchen.local.yml` and `.kitchen.local.yml` both being present.

Reply:
Hmm, I mean if it doesn't catch those I'm much less concerned, but let's see:

```ruby
localize = lambda path: path.sub(/(#{File.extname(path)})$/, '.local\1')
config_dir, config_base = File.split(config_file)
undot_config_base = config_base.gsub(/^\./, '')
dot_config_base = ".#{undot_config_base}"
if File.exist?(localize.call(File.join(config_dir, undot_config_base))) && File.exist?(localize.call(File.join(config_dir, dot_config_base)))
```

A little more verbose but maybe less magical with the regexes :)